src/daft-stats/src/partition_spec.rs

```diff
@@ -42,9 +42,19 @@ impl PartialEq for PartitionSpec {
         for field_name in self.keys.schema.as_ref().fields.keys() {
             let self_column = self.keys.get_column(field_name).unwrap();
             let other_column = other.keys.get_column(field_name).unwrap();
-            let value_eq = self_column.equal(other_column).unwrap().get(0).unwrap();
-            if !value_eq {
-                return false;
+            if let Some(value_eq) = self_column.equal(other_column).unwrap().get(0) {
+                if !value_eq {
+                    return false;
+                }
+            } else {
+                let both_null = self_column
```

Comment:
So for partition spec we want the behavior that `null == null`? if so, we should document that here